### PR TITLE
Admin app aesthetic fixes

### PR DESCRIPTION
**Pull request by ziakhan04**
Reviewer: @sidravi1 

Estimate: 10-20 minutes

----

# Aesthetic fixes to Admin App

## Ticket

Fixes: multiple tickets

## Description, Motivation and Context
Address the following:
- Add loading spinners to API calls
- Make navbar work in narrow screens via dropdown
- Set API token to a session only, so that if the wrong password is set, closing and re-opening the window resets the token and the user is asked for token again
- Navbar only has Manage Content and Playground. Easy to add other paths (like "Contact"), but not necessary right now. Only made things look unfinished.
- The screen automatically scrolls to the latest message in playground

## How Has This Been Tested?
- I tested by clicking around
- I also made sure that "npm run build" compiles without errors.

## To-do before merge
Click around yourself and test out the above new things.

## Checklist:

This checklist is a useful reminder of small things that can easily be forgotten.
Put an `x` in all the items that apply and remove any items that are not relevant to this PR.

- [x] My code follows the style guidelines of this project
- [x] I have reviewed my own code to ensure good quality
- [x] I have tested the functionality of my code to ensure it works as intended
- [x] I have resolved merge conflicts
- [ ] I have updated the automated tests (if applicable)
- [x] I have written [good commit messages][1]
- [ ] I have updated the README file (if applicable)
- [ ] I have updated affected documentation (if applicable)

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: admin_app/src/components/NavBar.tsx [hunk 1]
```diff
@@ -37,21 +46,18 @@ export const NavBar = () => {
             />
           </svg>
         </button>
-        <div className="hidden w-full md:block md:w-auto" id="navbar-default">
+        <div
```
Comment (sidravi1) [inline admin_app/src/components/NavBar.tsx:49]: Would be nice if (a) the menu icon shows that it is open (b) the menu floats on top of cards etc. and not take up space. But maybe these are for when we do styling with material-ui?

image

**Review by sidravi1 — APPROVED** [review]
Thanks @ziakhan04 - Looks good!
Approved with a few comments.

File: admin_app/src/app/chat/page.tsx [hunk 2]
```diff
@@ -26,6 +26,16 @@ export default function ChatPage() {
   }, [router, pathname]);
 
   const [messages, setMessages] = useState<Message[]>([]);
+  const endOfMessagesRef = useRef<null | HTMLDivElement>(null);
+
+  useEffect(() => {
```
[the remark this answers is not in the document]
Comment (ziakhan04) [reply admin_app/src/app/chat/page.tsx:31]: It is used to track the latest message div in terms of where it is.

File: admin_app/src/app/chat/page.tsx [hunk 3]
```diff
@@ -62,6 +72,7 @@ export default function ChatPage() {
               {messages.map((message, idx) => (
                 <ChatMessage key={idx} message={message} />
               ))}
+              <div ref={endOfMessagesRef} />
```
[the remark this answers is not in the document]
Comment (ziakhan04) [reply admin_app/src/app/chat/page.tsx:75]: So, whenever a new message comes in it autoscrolls to the latest message. I did it for two reasons -
- Mostly for convenience, I would be unsure if the message came or not and I had to always keep scrolling to see the latest message
- This is a common thing in most chat systems. ChatGPTs and Chatseed's UIs do the same
 
Can remove remove if it is annoying or find a better way in future.

File: admin_app/src/components/NavBar.tsx (hunk 1 above)
Comment (ziakhan04) [reply admin_app/src/components/NavBar.tsx:49]: Fixed (a). Could not fix (b) quick enough. I guess we can wait for the material refactor.

**Pull request merged**